static/js/forms.js

```diff
@@ -0,0 +1,23 @@
+/**
+ *
+ * @param {Node} submitButton
+ *
+ * Attaches a loading spinner to the submit button on
+ * form submission
+ */
+function attachLoadingSpinner(submitButton) {
+  const spinnerIcon = document.createElement("i");
+  spinnerIcon.className = "p-icon--spinner u-animation--spin is-light";
+  const buttonRect = submitButton.getBoundingClientRect();
+  submitButton.style.width = buttonRect.width + "px";
+  submitButton.style.height = buttonRect.height + "px";
+  submitButton.classList.add("is-processing");
+  submitButton.disabled = true;
+  submitButton.innerText = "";
+  submitButton.appendChild(spinnerIcon);
+}
+const form = document.querySelector("form");
+const submitButton = form.querySelector('button[type="submit"]');
```

Comment:
Just a small one, can we add an exclusion? e.g. "no-loader" or "no-spinner". There are certain situations where this fails, see [here the example I did](https://github.com/canonical/ubuntu.com/pull/13661/files).

The reason is that there is no end to the loading, and there are certain forms that don't submit to another page, so the spinner keeps rolling indefinitely.

Reply:
Added to the PR for these cases. Except surely this should be handled on the form-submission side, ie. the form should not attempt to submit (and trigger the spinner) until the user data has been validated?